tests/unit/test_dbt_runner.py

```diff
@@ -21,4 +23,8 @@ def test_invalid_command(self, dbt: dbtRunner) -> None:
             dbt.invoke(["invalid-command"])
 
     def test_invoke_version(self, dbt: dbtRunner) -> None:
-        dbt.invoke(["--version"])
+        try:
+            dbt.invoke(["--version"])
+        # click exits early for version, this is intended
+        except click.exceptions.Exit:
+            return
```

Comment:
Is this the best way to handle this exception? We do want to exit early when using version so no further commands are run, but putting this here felt a bit strange.

Reply:
Given that `dbtRunner` is meant for programmatic invocations - I think we should handle this in `dbtRunner` itself (as opposed to the tests) so that callers don't need to catch `click` exceptions. Perhaps in a new `except` block here: https://github.com/dbt-labs/dbt-core/compare/feature/click-cli#diff-43832707bfec8cdaa49c13f004d9b3296585827f7e68ef07fba7ad1d5e6eb7c6R55-R58